frontend/src/routes/profile/index.lazy.tsx

```diff
@@ -0,0 +1,53 @@
+import { $api } from "@/api/client";
+import ReservationCard from "@/components/reservation-card";
+import { createLazyFileRoute } from "@tanstack/react-router";
+
+export const Route = createLazyFileRoute("/profile/")({
+  component: RouteComponent,
+});
+
+function RouteComponent() {
+  const { data, isSuccess, isError } = $api.useQuery(
+    "get",
+    "/api/v1/reservations/",
+  );
+
+  if (isSuccess) {
+    return (
+      <div className="m-8 flex w-max flex-col gap-4">
+        <h1 className="text-xl font-bold">My reservations</h1>
+        {data && data.length == 0 && <h2>No reservations found</h2>}
+        {data &&
+          data.length > 0 &&
+          data.map((res) => {
```

Comment:
you can remove the first data check since ts is smart enough to know that when isSuccess is true the data is set
```suggestion
        {data.length === 0 ? <p>No reservations found</p> : data.map((res) => {
```